db/collection_index.go

```diff
@@ -107,6 +107,55 @@ func (db *db) fetchCollectionIndexDescriptions(
 	return indexDescriptions, nil
 }
 
+func (c *collection) CreateDocIndex(ctx context.Context, doc *client.Document) error {
+	txn, err := c.getTxn(ctx, false)
+	if err != nil {
+		return err
+	}
+	defer c.discardImplicitTxn(ctx, txn)
+
+	err = c.indexNewDoc(ctx, txn, doc)
+	if err != nil {
+		return err
+	}
+
+	return c.commitImplicitTxn(ctx, txn)
+}
+
+func (c *collection) UpdateDocIndex(ctx context.Context, oldDoc, newDoc *client.Document) error {
+	txn, err := c.getTxn(ctx, false)
+	if err != nil {
+		return err
+	}
+	defer c.discardImplicitTxn(ctx, txn)
+
+	err = c.deleteIndexedDoc(ctx, txn, oldDoc)
+	if err != nil {
+		return err
+	}
+	err = c.indexNewDoc(ctx, txn, newDoc)
```

Comment:
nitpick: I do see `delete+new` as still being business logic, but only at a nitpick level - so feel free to merge as is :)

Reply:
Can you expand on what is your nitpick concern here Andy?

Reply:
I have none and thought I was in the txn_db.go file :)

Reply:
Ah I see I how I got confused - they have moved to collection, and I new that they had moved, but my brain still thought of the implementation as being in the txn_db file when I saw the txn logic.  I wonder if we can do a similar job to collection that we did to client.Store and split out the txn boilerplate to another file sometime